server/properties.lua

```diff
@@ -0,0 +1,228 @@
+    if not result then return end
+
+    for _, v in pairs(result) do
+        local propertyowners = MySQL.query.await('SELECT * FROM property_owners WHERE property_id = ?', { v.id })
+        local owners = {}
+
+        if propertyowners then
+            for _, owner in pairs(propertyowners) do
+                owners[owner.citizenid] = (owner.owner and 'owner') or (owner.co_owner and 'co_owner') or
+                (owner.tenant and 'tenant') or false
+            end
+        end
+
+        v.coords = json.decode(v.coords)
+        properties[v.id] = {
+            name = v.name,
+            interior = v.interior,
+            furnished = v.furnished or false,
+            decorations = v.decorations or nil,
+            garage = v.garage or false,
+            garage_slots = (v.garage_slots and json.decode(v.garage_slots)) or nil,
+            coords = vector4(v.coords.x, v.coords.y, v.coords.z, v.coords.h),
+            stash = v.stash or nil,
+            logout = v.logout or nil,
+            outfit = v.outfit or nil,
+            price = v.price,
+            rent = v.rent,
+            rent_date = v.rent_date ~= 0 and v.rent_date/1000 or false,
+            owners = next(owners) and owners or false,
+            inside = {}
+        }
+    end
+
+    for k, v in pairs(properties) do
+        local propertyCoords = v.coords
+        local found = false
+        for _, group in pairs(propertiesGroups) do
+            local calcCoords = vec3(math.floor(propertyCoords.x + 0.5), math.floor(propertyCoords.y + 0.5), math.floor(propertyCoords.z + 0.5))
+            local calcGroupCoords = vec3(math.floor(group.coords.x + 0.5), math.floor(group.coords.y + 0.5), math.floor(group.coords.z + 0.5))
+
+            if calcGroupCoords == calcCoords then
+                group.properties[#group.properties + 1] = k
+                found = true
+                break
+            end
+        end
+        if not found then
+            propertiesGroups[#propertiesGroups + 1] = {
+                name = v.name,
+                coords = propertyCoords,
+                properties = {k},
+                propertyType = v.garage and 'garage' or 'property'
+            }
+        end
+    end
+
+    fillInsideTable()
+
+    TriggerClientEvent('qbx-property:client:refreshProperties', -1)
+end
+
+-- Enter furnished property
+RegisterNetEvent('qbx-property:server:enterProperty', function(propertyid)
+    local source = source
+    local concealedPlayers = {}
+    local PlayersInside = {}
+
+    for k, v in pairs(properties) do
+        if v.garage or not v.furnished then goto continue end
+        if k == propertyid then
+            for _, serverid in pairs(v.inside) do
+                PlayersInside[#PlayersInside + 1] = serverid
+            end
+            goto continue
+        end
+        for _, serverid in pairs(v.inside) do
+            concealedPlayers[#concealedPlayers + 1] = serverid
+        end
+        ::continue::

```

Comment:
Consider extracting functions over using goto

Reply:
modified it to not used goto (simpler than extracting two 3 line functions)